include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -78,6 +80,104 @@ __find_start_point(const _Rng1& __rng1, const _Rng2& __rng2, const _Index __i_el
     }
 }
 
+//Searching for an intersection of a merge matrix (n1, n2) diagonal with the Merge Path to define sub-ranges
+//to serial merge. For example, a merge matrix for [0,1,1,2,3] and [0,0,2,3] is shown below:
+//     0   1  1  2   3
+//    ------------------
+//   |--->
+// 0 | 0 | 1  1  1   1
+//   |   |
+// 0 | 0 | 1  1  1   1
+//   |   ---------->
+// 2 | 0   0  0  0 | 1
+//   |             ---->
+// 3 | 0   0  0  0   0 |
+template <typename _Rng1, typename _Rng2, typename _Index, typename _Compare>
+_split_point_t<_Index>
+__find_start_point_in(const _Rng1& __rng1, const _Index __rng1_from, _Index __rng1_to, const _Rng2& __rng2,
```

Comment:
It seems like we should be able to use the existing `__find_start_point` simply by restricting the ranges via `drop_view_simple`, by their `__rng*_from` value, passing in an appropriate `_n1` and `__n2` to match the tile size, and fixing `__i_elem` to be relative to the previous bounding base diagonal.  
Similar to what is done in merge sort here https://github.com/oneapi-src/oneDPL/blob/7150d5c26dab8de50052f6c5e5064a229203b989/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h#L270-L278 .

I would be surprised / interested to understand why if that is worse performing than this additional implementation.

Reply:
I don't think so:
1) the `oneapi::dpl::__ranges::drop_view_simple` has his own implementation problems: for example it's returns incorrect `size()` result - it's returns the size of initial range.
2) please see my comment at https://github.com/oneapi-src/oneDPL/pull/1933#issuecomment-2556658545 - I rechecked and think that `__find_start_point_in` is ok now.